chain/jsonrpc-primitives/src/types/transactions.rs

```diff
@@ -5,17 +5,19 @@ pub struct RpcBroadcastTransactionRequest {
     pub signed_transaction: near_primitives::transaction::SignedTransaction,
 }
 
-#[derive(Debug)]
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
 pub struct RpcTransactionStatusCommonRequest {
+    #[serde(flatten)]
```

Comment:
I want to be able to pass to RPC node params
```
{
    "tx_hash": "6zgh2u9DqHHiXzdy9ouTP7oGky2T4nugqzqt9wJZwNFm", 
    "sender_account_id": "sender.testnet"
}
```
Without flatten, I'll have to pass
```
{
"transaction_info": {
    "tx_hash": "6zgh2u9DqHHiXzdy9ouTP7oGky2T4nugqzqt9wJZwNFm", 
    "sender_account_id": "sender.testnet"
}
}
```

So it flattens the object for (de)serialisation

Reply:
Why do you want the flat representation? If only for convenience I would argue it's better to do the more verbose version. It's more intuitive, more readable and has better compatibility prospects for the future. If we ever want to add more fields to the `RpcTransactionStatusCommonRequest` it will be much easier with non-flattened structure.

Reply:
It's not the thing for the core devs. It's the thing for external users. They don't care about core structures, 99.999...% have never seen them and actually it's great. 
They just want to pass the required fields and receive the response.
So I am sure we need to make that json as simple as possible.

Moreover, these Rust structures may become even more complicated, we may want to nest them even more, add fields, etc. It's better not to rely on a snapshot of current structures as the backward compatible interface for the external users.